deltacat/compute/compactor/compaction_session.py

```diff
@@ -454,14 +535,54 @@ def _execute_compaction_round(
         deltacat_storage=deltacat_storage,
     )
     logger.info(f"Getting {len(mat_tasks_pending)} materialize result(s)...")
-    mat_results = ray.get(mat_tasks_pending)
+    mat_results: List[MaterializeResult] = ray.get(mat_tasks_pending)
     logger.info(f"Got {len(mat_results)} materialize result(s).")
 
-    log_current_cluster_utilization(log_identifier="post_materialize")
+    materialize_end = time.time()
+
+    cluster_utilization_after_mat, cluster_util_after_mat_latency = timed_invocation(
+        ClusterUtilization.get_current_cluster_utilization
+    )
+
+    compaction_audit.set_materialize_time_in_seconds(
+        materialize_end - materialize_start
+    )
 
     mat_results = sorted(mat_results, key=lambda m: m.task_index)
     deltas = [m.delta for m in mat_results]
 
+    telemetry_time_materialize = sum(
+        [mat_result.telemetry_time_in_seconds for mat_result in mat_results]
+    )
+    peak_memory_usage_materialize = max(
+        [mat_result.peak_memory_usage_bytes for mat_result in mat_results]
+    )
+
+    compaction_audit.set_peak_memory_used_bytes_by_materialize(
+        peak_memory_usage_materialize.item()
+    )
+    compaction_audit.set_total_object_store_memory_used_bytes(
+        cluster_utilization_after_mat.used_object_store_memory_bytes
+    )
+    compaction_audit.set_peak_memory_used_bytes(
+        max(
+            [
+                compaction_audit.peak_memory_used_bytes_by_hash_bucketing,
+                compaction_audit.peak_memory_used_bytes_by_dedupe,
+                compaction_audit.peak_memory_used_bytes_by_materialize,
+            ]
+        )
+    )
+
+    compaction_audit.set_telemetry_time_in_seconds(
+        cluster_util_after_hb_latency
+        + cluster_util_after_dd_latency
+        + telemetry_time_dd
+        + telemetry_time_hb
+        + telemetry_time_materialize
+        + cluster_util_after_mat_latency
+    )
+
```

Comment:
As long as we're saving or at least capturing a sum of all of the telemetry collection latencies in the above recommended change to create a higher-level method to save step stats, then we can also roll this block + total compaction time + pyarrow write result persistence into a higher level method like `compaction_audit.save_round_completion_stats(total_obj_store_mem_used, pyarrow_write_result, compaction_end - compaction-start)`

I'm also assuming here that we'll take the max of all peak memory usage and the sum of all telemetry latencies previously collected inside of this method.